Allow to define block and allow lists for providers (#415)
jupyterlab/jupyter-ai: docs/source/users/index.md

The `--request-schema` parameter is the JSON object the endpoint expects as input, with the prompt being substituted into any value that matches the string literal `"<prompt>"`. For example, the request schema `{"text_inputs":"<prompt>"}` will submit a JSON object with the prompt stored under the `text_inputs` key.

The `--response-path` option is a [JSONPath](https://goessner.net/articles/JsonPath/index.html) string that retrieves the language model's output from the endpoint's JSON response. For example, if your endpoint returns an object with the schema `{"generated_texts":["<output>"]}`, its response path is `generated_texts.[0]`.

> ## Configuration

Reviewer: Should we add an introduction to this section?

```suggestion
## Configuration

You can specify an allowlist, to only allow a certain list of providers, or a blocklist, to block some providers.
```

Can uses specify both?

Reviewer 2: Yes, both should be applicable at one time.

> ### Block-listing providers

Reviewer: ```suggestion
### Blocklisting providers
```

This configuration allows for blocking specific providers in the settings panel. This list takes precedence over the allowlist in the next section.

```
jupyter lab --Ai.blocked_providers=openai
```

To block more than one provider in the block-list, repeat the runtime configuration.

```
jupyter lab --Ai.blocked_providers=openai --Ai.blocked_providers=ai21
```

> ### Allow-listing providers

Reviewer: ```suggestion
### Allowlisting providers
```

> This configuration allows for filtering the list of providers in the settings panel to only an allow-listed set of providers.

Reviewer: ```suggestion
This configuration allows for filtering the list of providers in the settings panel to only an allowlisted set of providers.
```

```
jupyter lab --Ai.allowed_providers=openai
```

> To allow more than one provider in the allow-list, repeat the runtime configuration.

Reviewer: ```suggestion
To allow more than one provider in the allowlist, repeat the runtime configuration.
```

The `--request-schema` parameter is the JSON object the endpoint expects as input, with the prompt being substituted into any value that matches the string literal `"<prompt>"`. For example, the request schema `{"text_inputs":"<prompt>"}` will submit a JSON object with the prompt stored under the `text_inputs` key.

The `--response-path` option is a [JSONPath](https://goessner.net/articles/JsonPath/index.html) string that retrieves the language model's output from the endpoint's JSON response. For example, if your endpoint returns an object with the schema `{"generated_texts":["<output>"]}`, its response path is `generated_texts.[0]`.

## Configuration

You can specify an allowlist, to only allow a certain list of providers, or a blocklist, to block some providers.

> ### Blocklisting providers

Reviewer: Should these be in alphabetical order? (Allowlist, blocklist)

Reviewer 2: I added blocklist first because they take precedence in the configuration.

This configuration allows for blocking specific providers in the settings panel. This list takes precedence over the allowlist in the next section.

```

> jupyter lab --Ai.blocked_providers=openai

Author: ```suggestion
jupyter lab --AiExtension.blocked_providers=openai
```

Reviewer 2: Corrected, thanks for catching that.

```

To block more than one provider in the block-list, repeat the runtime configuration.

```

> jupyter lab --Ai.blocked_providers=openai --Ai.blocked_providers=ai21

Author: ```suggestion
jupyter lab --AiExtension.blocked_providers=openai --AiExtension.blocked_providers=ai21
```

```

### Allowlisting providers
This configuration allows for filtering the list of providers in the settings panel to only an allowlisted set of providers.

```

> jupyter lab --Ai.allowed_providers=openai

Author: ```suggestion
jupyter lab --AiExtension.allowed_providers=openai
```

```

To allow more than one provider in the allowlist, repeat the runtime configuration.

```

> jupyter lab --Ai.allowed_providers=openai --Ai.allowed_providers=ai21

Author: ```suggestion
jupyter lab --AiExtension.allowed_providers=openai --AiExtension.allowed_providers=ai21
```
